tests/e2e/scale/noobaa/test_list_objects.py

```diff
@@ -98,3 +103,66 @@ def test_list_large_number_of_objects(
             f"Files count {files_count} and objects_count {objects_count} does not match. "
             "Check does all objects are listed correctly!!!"
         )
+
+    @pytest.fixture()
+    def setup_dir(self, request, awscli_pod_session, test_directory_setup):
+        """
+        This is to setup 1000s of directories where in each directory has atleast 10 objects
+        and one directory contains almost 3 Mil objects
+        """
+        origin_dir = test_directory_setup.origin_dir
+
+        # create 1000 directories under origin_dir
+        amount = 1000
+        command = (
+            f"for i in $(seq 0 {amount-1});"
+            f"do mkdir {origin_dir}/dir_$i && chmod -R 777 {origin_dir}/dir_$i; done"
+        )
+        awscli_pod_session.exec_sh_cmd_on_pod(command=command, sh="sh", timeout=9000)
```

Comment:
Don't think we would need 9k sec timeout. Not sure though, Alter based on your testing